test/unit/compiler/passes/merge-character-classes.spec.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+const chai = require("chai");
+const helpers = require("./helpers");
+const pass = require("../../../../lib/compiler/passes/merge-character-classes");
+
+chai.use(helpers);
+
+const expect = chai.expect;
+
+describe("compiler pass |mergeCharacterClasses|", () => {
+  it("Merges single character literals, class, and ref_rules", () => {
+    expect(pass).to.changeAST(
+      [
+        "one = three / $('a' / 'd') / [c-f] / [efh] / [c-h] / [c-g]",
+        "two = 'P' / 'P' / three / 'P' / [Q-T]",
+        "three = $('x' / [u-w])",
+        "four = 'a' / [aaa] / 'a'",
+      ].join("\n"),
+      {
+        rules: [
+          {
+            name: "one",
+            expression: {
+              type: "class",
+              parts: ["a", ["c", "h"], ["u", "x"]],
+              ignoreCase: false,
+              inverted: false,
+              location: {
+                start: { line: 1, column: 7 },
+                end: { line: 1, column: 59 },
+              },
+            },
+          },
+          {
+            name: "two",
+            expression: {
+              type: "class",
+              parts: [["P", "T"], ["u", "x"]],
+              ignoreCase: false,
+              inverted: false,
+              location: {
+                start: { line: 2, column: 7 },
+                end: { line: 2, column: 38 },
+              },
+            },
+          },
+          {
+            name: "three",
+            expression: {
+              type: "class",
+              parts: [["u", "x"]],
+              ignoreCase: false,
+              inverted: false,
+              location: {
+                start: { line: 3, column: 11 },
+                end: { line: 3, column: 22 },
+              },
```

Comment:
This location covers
```pegjs
'x' / [u-w]
```
Shouldn't it cover
```pegjs
$('x' / [u-w])
```
?

Reply:
will fix